src/uwtools/api/config.py

```diff
@@ -28,24 +33,50 @@ def compare(
 
 
 def realize(
-    input_file: OptionalPath,
-    input_format: str,
-    output_file: OptionalPath,
-    output_format: str,
-    values_file: OptionalPath,
-    values_format: Optional[str],
+    input_config: Union[dict, _Config, OptionalPath] = None,
+    input_format: Optional[str] = None,
+    output_file: OptionalPath = None,
+    output_format: Optional[str] = None,
+    values: Union[dict, _Config, OptionalPath] = None,
+    values_format: Optional[str] = None,
     values_needed: bool = False,
     dry_run: bool = False,
 ) -> bool:
```

Comment:
For the API analog to `uw config realize`, we now support providing the "base" input config as either a `dict` or some kind of `Config` object, in addition to existing support for filesystem paths and reads from `stdin`. Similarly for the values. (I'm not sure if `OptionalPath` is the correct type for the values argument, but this argument will go away soon due to [UW-421](https://jira.epic.oarcloud.noaa.gov/browse/UW-421).

The argument names changed to reflect the fact that we are not necessarily dealing with `file`s now. I would have preferred `input` over `input_config`, but that conflicts with a Python built-in. I'm open to better ideas about any names!

I also added some `None` defaults so that API users need not specify optional arguments. I didn't notice the need before because only the CLI is currently calling these API functions, and it supplies values for _all_ arguments. But API users should not have to.